<launch>
<arg name="Imu_name" default="Xsens" />
<arg name="suffix" default="" />
<arg name="onboard_capture" default="true" />
<arg name="ros_madgwick_capture" default="false" />
<arg name="ros_madgwick_corrected_capture" default="false" />
<arg name="freq" default="100" />
<arg name="launch_driver" default="true" />

  <group if="$(arg launch_driver)">
  <!-- start driver and visualization -->
    <include file="$(find sensor_imu_mti_xsens)/launch/xsens_mti.launch"> </include>
    <param name="xsens_mti/frequency" value="$(arg freq)" />
  </group>
  
  <!-- recording of on board algorithm -->
  <group if="$(arg onboard_capture)">
    <node pkg="imu_calibration" type="capture_imu_data.py" name="$(arg Imu_name)_onboard_capture" output="screen">
      <param name="file_name_of_data" value="$(arg Imu_name)_data_onboard$(arg suffix).npy" />
      <param name="path_of_file_save" value="$(find imu_calibration)/data/imu_recordings/" />
      <param name="imu_subscription_topic_name" value="/sensor/imu/xsens_mti/data" />
      <param name="mag_subscription_topic_name" value="/sensor/imu/xsens_mti/mag_calib" />
    </node>
  </group>

  <!-- recording of madgwick ros algorithm -->
  <group if="$(arg ros_madgwick_capture)">
    <node pkg="imu_calibration" type="capture_imu_data.py" name="$(arg Imu_name)_madgwick_capture" output="screen">
      <param name="file_name_of_data" value="$(arg Imu_name)_data_madgwick$(arg suffix).npy" />
      <param name="path_of_file_save" value="$(find imu_calibration)/data/imu_recordings/" />
      <param name="imu_subscription_topic_name" value="madgwick_imu_filter_node/$(arg Imu_name)/imu" />
      <param name="mag_subscription_topic_name" value="/sensor/imu/xsens_mti/mag_calib" />
    </node>
  </group>

      <!-- recording of madgwick corrected ros algorithm -->
  <group if="$(arg ros_madgwick_corrected_capture)">
    <node pkg="imu_calibration" type="capture_imu_data.py" name="$(arg Imu_name)_madgwick_corrected_capture" output="screen">
      <param name="file_name_of_data" value="$(arg Imu_name)_data_madgwick_corrected$(arg suffix).npy" />
      <param name="path_of_file_save" value="$(find imu_calibration)/data/imu_recordings/" />
      <param name="imu_subscription_topic_name" value="madgwick_imu_filter_corrected_node/$(arg Imu_name)/imu" />
      <param name="mag_subscription_topic_name" value="/sensor/imu/xsens_mti/mag_calib" />
    </node>
  </group>

</launch>
